### ISSUE TITLE
File with which Multivalent fails with: "psotmp.1517.conv.mi.tmp.pdf: java.lang.ClassCastException: com.pt.doc.PostScript$1 cannot be cast to multivalent.std.adaptor.pdf.Dict"

### ISSUE THREAD

**Issue by rbrito** [pts/pdfsizeopt#56]
Dear @pts,

You told me before that you wanted to know about some files that are able to crash Multivalent, since you may have an idea of how to workaround some failures of Multivalent and, thus, get better compression.

I have some files where Multivalent fails for many different reasons, but one of the most common errors is with the message above. The complete message is the following:

```
info: This is pdfsizeopt rUNKNOWN size=382141.
info: prepending to PATH: /home/rbrito/Downloads/pdfsizeopt
info: using Multivalent.jar: /home/rbrito/Downloads/pdfsizeopt/Multivalent.jar
info: using Java for Multivalent: /usr/bin/java
info: using Multivalent compress command: /usr/bin/java -cp /home/rbrito/Downloads/pdfsizeopt/Multivalent.jar -Djava.awt.headless=true tool.pdf.Compress
info: loading PDF from: 1206.3686v1.pdf
info: loaded PDF of 226558 bytes
info: separated to 220 objs + xref + trailer
info: parsed 220 objs
info: found 0 Type1 fonts loaded
info: found 27 Type1C fonts loaded
info: writing Type1CParser (64703 font bytes) to: psotmp.1517.conv.parse.tmp.ps
info: using Ghostscript /usr/bin/gs: GPL Ghostscript 9.22 (2017-10-04)
info: executing Type1CParser with Ghostscript: gs -q -P- -dNOPAUSE -dBATCH -sDEVICE=nullpage -sDataFile=psotmp.1517.conv.parsedata.tmp.ps -f psotmp.1517.conv.parse.tmp.ps
Type1CParser: using interpreter GPL Ghostscript 922 20171004
Type1CParser: all OK
info: parsed 27 Type1C fonts
info: writing Type1CGenerator (165430 bytes in 27 fonts) to: psotmp.1517.conv.gen.tmp.ps
info: executing Type1CGenerator with Ghostscript: gs -q -P- -dNOPAUSE -dBATCH -sDEVICE=pdfwrite -dPDFSETTINGS=/printer -dColorConversionStrategy=/LeaveColorUnchanged -sOutputFile=psotmp.1517.conv.gen.tmp.pdf -f psotmp.1517.conv.gen.tmp.ps
Type1CGenerator: using interpreter GPL Ghostscript 922 20171004
Type1CGenerator: all OK
info: loading PDF from: psotmp.1517.conv.gen.tmp.pdf
info: loaded PDF of 89034 bytes
info: separated to 122 objs + xref + trailer
info: parsed 122 objs
info: found 27 fonts loaded
info: optimized Type1C fonts to form 71037 bytes to 70942 bytes (100%)
info: optimized 44 streams, kept 21 #orig, 10 uncompressed, 13 zip
info: eliminated 15 duplicate objs
info: eliminated 12 unused objs in 12 classes
info: saving PDF with 193 objs with Multivalent to: 1206.3686v1.psom.pdf
info: writing Multivalent input PDF: psotmp.1517.conv.mi.tmp.pdf
info: generated object stream of 6912 bytes in 149 objects (12%)
info: written 191999 bytes to Multivalent input PDF: psotmp.1517.conv.mi.tmp.pdf
info: executing Multivalent to optimize PDF: /usr/bin/java -cp /home/rbrito/Downloads/pdfsizeopt/Multivalent.jar -Djava.awt.headless=true tool.pdf.Compress -nopagepiece -noalt -mon psotmp.1517.conv.mi.tmp.pdf
psotmp.1517.conv.mi.tmp.pdf: java.lang.ClassCastException: com.pt.doc.PostScript$1 cannot be cast to multivalent.std.adaptor.pdf.Dict
info: Multivalent has not created output: psotmp.1517.conv.mi.tmp-o.pdf
Traceback (most recent call last):
  File "/home/rbrito/Downloads/pdfsizeopt/pdfsizeopt", line 41, in <module>
    sys.exit(main.main(sys.argv, script_dir=script_dir))
  File "/home/rbrito/Downloads/pdfsizeopt/lib/pdfsizeopt/main.py", line 9218, in main
    not f.do_decompress_most_streams))
  File "/home/rbrito/Downloads/pdfsizeopt/lib/pdfsizeopt/main.py", line 8767, in Save
    multivalent_compress_command=multivalent_compress_command)
  File "/home/rbrito/Downloads/pdfsizeopt/lib/pdfsizeopt/main.py", line 8690, in _RunMultivalent
    assert False, 'Multivalent failed (no output)'
AssertionError: Multivalent failed (no output)
```

Here is one example that triggers it: [1206.3686v1.pdf](https://github.com/pts/pdfsizeopt/files/1378073/1206.3686v1.pdf)

Thanks,

Rogério.

**Comment by pts** [comment]
Thank you for reporting this!

Fixing and working around Multivalent issues is low priority for me now. But it's nice to have the offending input data collected.

**Comment by rbrito** [comment]
Sure, very low priority. No problems with that.

**Comment by pts** [comment]
Fixed in 69ec4c7fd9988e87e6f5ce5fd18868e02bfa1314.

Oddly enough, Multivalent fails if the `/Type /XRef` obj has `/Predictor 10`. It seems to work with `/Predictor 12` and `/Predictor 2` though. (This is a bug in Multivalent, because according to pdf_reference_1-7.pdf, `/Predictor 10` and `/Predictor 12` are equivalent for decompression.)
